drivers/src/gicv2.rs

```diff
@@ -0,0 +1,255 @@
+use tock_registers::interfaces::{ReadWriteable, Readable, Writeable};
+use tock_registers::register_bitfields;
+use tock_registers::registers::{ReadOnly, ReadWrite};
+
+pub struct GicV2 {
+    pub distributor: &'static GicDistributor,
+    pub cpu: &'static GicCpu,
+}
+
+impl GicV2 {
+    pub fn new(distributor_base: usize, cpu_base: usize) -> Self {
+        let distributor = unsafe { &*(distributor_base as *const GicDistributor) };
+        let cpu = unsafe { &*(cpu_base as *const GicCpu) };
+        let mut gic = Self { distributor, cpu };
+
+        gic.init_distributor();
+
+        gic
+    }
+
+    pub fn nlines(&self) -> usize {
+        let n = self.distributor.TYPER.read(GICD_TYPER::ITLinesNumber) as usize;
+
+        32 * (n + 1)
+    }
+
+    pub fn sec_ext(&self) -> u32 {
+        self.distributor.TYPER.get() & (1 << 10)
+    }
+
+    /// Put the Gic in a known state.
+    fn init_distributor(&mut self) {
+        self.distributor.CTLR.modify(GICD_CTLR::EnableGrp0::Disable);
+        self.distributor.CTLR.modify(GICD_CTLR::EnableGrp1::Disable);
+
+        // Disable all interrupts.
+        for i in 0..(self.nlines() / 32) {
+            self.distributor.ICENABLER[i].set(0xffff_ffff);
```

Comment:
It's not really a magic value, we are disabling interrupts by putting 1 everywhere in the clear-enable register. I could put a comment

Reply:
All the occurences of `0xffff_ffff` don't mean _interrupt disable_.
So I would have to create 3 constants, which would be used nowhere else.
And it would be easier to check if the value is correct right where it it used.

Reply:
Fair enough